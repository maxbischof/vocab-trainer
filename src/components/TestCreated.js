import React from 'react'
import Link from './ui/Link'
import styled from 'styled-components'
import Button from './ui/Button'
import { CopyToClipboard } from 'react-copy-to-clipboard'

export default function TestCreated({ testURL }) {
  const fullURL = window.location.host + testURL
  return (
    <StyledTestLink>
      <h1>Test erstellt</h1>
      Dein Link zum Test:
      <Link href={testURL}>{fullURL}</Link>
      <CopyToClipboard text={fullURL}>
        <Button>Copy</Button>
      </CopyToClipboard>
    </StyledTestLink>
  )
}

const StyledTestLink = styled.div`
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
`
