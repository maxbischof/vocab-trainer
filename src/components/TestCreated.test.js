import React from 'react'
import TestCreated from './TestCreated'
import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const testURL = '/tests/123456'
  const tree = renderer
    .create(
      <Router>
        <TestCreated testURL={testURL} />
      </Router>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
