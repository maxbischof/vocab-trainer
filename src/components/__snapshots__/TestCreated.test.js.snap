// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-dlfnbm jyuGPq"
>
  <h1>
    Test erstellt
  </h1>
  Dein Link zum Test:
  <a
    className="sc-bdfBwQ iGLRAW"
    href="/tests/123456"
    onClick={[Function]}
  >
    localhost/tests/123456
  </a>
  <button
    className="sc-gsTCUz hobRdM"
    onClick={[Function]}
  >
    Copy
  </button>
</div>
`;
